app/src/main/java/org/mozilla/fenix/home/pocket/PocketCategoriesViewHolder.kt

```diff
@@ -49,11 +49,17 @@ class PocketCategoriesViewHolder(
             composeView.resources.getDimensionPixelSize(R.dimen.home_item_horizontal_margin)
         composeView.setPadding(horizontalPadding, 0, horizontalPadding, 0)
 
+        val homeScreenReady = components.appStore
+            .observeAsComposableState { state -> state.firstFrameDrawn }.value ?: false
+
         val categories = components.appStore
             .observeAsComposableState { state -> state.pocketStoriesCategories }.value
         val categoriesSelections = components.appStore
             .observeAsComposableState { state -> state.pocketStoriesCategoriesSelections }.value
 
+        // Perceived performance can be improved by waiting to draw pocket content until after the
+        // first frame is drawn. See https://github.com/mozilla-mobile/fenix/issues/25545 for details.
+        if (!homeScreenReady) return
```

Comment:
why not instead just wrap the `Column` with this `if`?

Reply:
Less indentation I guess? I've usually just prefer early returns, stylistically. There's additionally reasoning for the choice in the other comment, and I think it would be good to keep the two in sync if we decide to change it.